base/linalg/exceptions.jl

```diff
@@ -11,7 +11,18 @@ type LAPACKException <: Exception
 end
 
 type ARPACKException <: Exception
-    info::BlasInt
+    info::ByteString
```

Comment:
Yes. That's true. We could backport the promotion commit only, or assume that nobody relies on the internals of `ARPACKException` which is probably a safe assumption. They are thrown very rarely because we capture most of the errors with checks before calling ARPACK.